### ISSUE TITLE
DynamicForm item tab order should be consistent when using CanvasItem

### ISSUE THREAD

**Issue by GoogleCodeExporter** [isomorphic-software/smartgwt#653]
```
What steps will reproduce the problem?
1. Create a DynamicForm
2. Add 2 TextItems
3. Add a CanvasItem with 2 IButtons

What is the expected output? What do you see instead?
I expect that the tab order would be:
1. TextItem #1
2. TextItem #2
3. IButton #1
4. IButton #2

Instead it is:
1. IButton #1
2. TextItem #1
3. IButton #2
4. TextItem #2


What version of the product are you using? On what operating system?
SmartGWT 3.0
3.0p from 2012-01-22
GWT 2.3.0
Win7


What browser(s) does this happen in?  Are there any browsers where the
issue does not occur?
FireFox 6.0.2
Chrome 16.0.912.75 m


Please provide any additional information below.
This works fine in SmartGWT 2.5
```

Original issue reported on code.google.com by `shortpa...@gmail.com` on 22 Jan 2012 at 8:59

Attachments:
- [TestFormItemTabOrder.java](https://storage.googleapis.com/google-code-attachments/smartgwt/issue-653/comment-0/TestFormItemTabOrder.java)
